salt/api-gateway/config/etc-kong-nginx_kong.lua

```diff
@@ -64,28 +60,19 @@ upstream kong_upstream {
     keepalive ${{UPSTREAM_KEEPALIVE}};
 }
 
-map $http_upgrade $upstream_connection {
-    default keep-alive;
-    websocket upgrade;
-}
-
-map $http_upgrade $upstream_upgrade {
-    default '';
-    websocket websocket;
-}
-
 server {
     server_name kong;
-    listen ${{PROXY_LISTEN}};
-    error_page 404 408 411 412 413 414 417 /kong_error_handler;
+    listen ${{PROXY_LISTEN}}${{PROXY_PROTOCOL}};
```

Comment:
massive configuration file containing code. Should we actually maintain a copy of this in the formula if it's that complicated?

Reply:
if has the elife mega-hack in it. if we can find a better home for that, perhaps in an addon, we can get rid of the custom configuration and stop maintaining this file.

with the upgraded kong we also get access to improvements in the addons as well.